### PR TITLE
Updating embedded deactivation process

### PR DESCRIPTION
**Pull request by rubenzorrilla**
Now the "full" negative nodes (those that do not belong to a split element) are deactivated and their velocity and pressure values are set to zero.

I took the change to separate the process in a header and a source file. Besides I add the possibility of setting the distance "bad cut" settings in the ProjectParameters.json.

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+}
+
+DistanceModificationProcess::DistanceModificationProcess(
+    ModelPart& rModelPart,
+    Parameters& rParameters)
+    : Process(), mrModelPart(rModelPart) {
+
+    Parameters default_parameters( R"(
+    {
+        "mesh_id"                                : 0,
```
Comment (RiccardoRossi) [inline applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp]: you don t need the meshid

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    mNegElemDeactivation = rParameters["deactivate_full_negative_elements"].GetBool();
+    mRecoverOriginalDistance = rParameters["recover_original_distance_at_each_step"].GetBool();
+}
+
+void DistanceModificationProcess::ExecuteInitialize() {
+
+    KRATOS_TRY;
+
+    // Variables check
+    if( mrModelPart.NodesBegin()->SolutionStepsDataHas( DISTANCE ) == false )
```
Comment (RiccardoRossi) [inline applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp]: you could use the error macros here

File: applications/FluidDynamicsApplication/python_scripts/apply_distance_modification_process.py [hunk 3]
```diff
@@ -15,11 +15,13 @@ def __init__(self, Model, settings):
 
         default_parameters = KratosMultiphysics.Parameters( """
         {
-            "mesh_id"                                   : 0,
-            "model_part_name"                           : "CHOOSE_FLUID_MODELPART_NAME",
-            "check_at_each_time_step"                   : false,
-            "deactivate_full_negative_elements"         : true,
-            "recover_original_distance_at_each_step"    : false
+            "mesh_id"                                : 0,
```
Comment (RiccardoRossi) [inline applications/FluidDynamicsApplication/python_scripts/apply_distance_modification_process.py]: take out the mesh_id

**Review by RiccardoRossi — DISMISSED** [review]
minor comments. approving otherwise

File: applications/FluidDynamicsApplication/python_scripts/apply_distance_modification_process.py (hunk 3 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/apply_distance_modification_process.py]: Done (in the interface as well).

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp (hunk 1 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp]: Removed.

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp (hunk 2 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp]: Done!

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**